src/script/lua_api/l_object.cpp

```diff
@@ -684,6 +684,85 @@ int ObjectRef::l_get_properties(lua_State *L)
 	return 1;
 }
 
+// set_observers(self, observers)
+int ObjectRef::l_set_observers(lua_State *L)
+{
+	GET_ENV_PTR;
+	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == nullptr)
+		throw LuaError("Invalid ObjectRef");
+
+	// Reset object to "unmanaged" (sent to everyone)?
+	if (lua_isnoneornil(L, 2)) {
+		ServerActiveObject *parent = sao->getParent();
+		if (parent != nullptr && !parent->m_observer_names.has_value())
```

Comment:
isn't the condition the wrong way around?
if the parent *has* observers, then you can't *remove* the childs observers

Reply:
Yep, fixed